contracts/contracts/solidity/StarknetValidator.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorValidatorInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/TypeAndVersionInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AccessControllerInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol';
+import '@chainlink/contracts/src/v0.8/SimpleWriteAccessController.sol';
+
+import '@chainlink/contracts/src/v0.8/dev/vendor/openzeppelin-solidity/v4.3.1/contracts/utils/Address.sol';
+
+import '../../vendor/starkware-libs/starkgate-contracts-solidity-v0.8/contracts/starkware/starknet/solidity/IStarknetMessaging.sol';
+
+/**
+ * @title StarkNetValidator - makes cross chain call to update the Sequencer Uptime Feed on L2
+ */
+contract StarkNetValidator is TypeAndVersionInterface, AggregatorValidatorInterface, SimpleWriteAccessController {
+    int256 private constant ANSWER_SEQ_OFFLINE = 1;
+    /* Selector hardcoded because StarkNet generates selectors differently than the standard ethereum way
+    different hash function on stark curve is used */
+    uint256 constant STARK_SELECTOR_UPDATE_STATUS =
+        1585322027166395525705364165097050997465692350398750944680096081848180365267;
+
+    IStarknetMessaging public immutable STARKNET_MESSAGING;
+    uint256 public immutable L2_UPTIME_FEED_ADDR;
+
+    /** 
+     * Invalid StarkNet Messaging address.
+     * The address is 0.
+     * @param starkNetMessagingAddr address.
+     */
+    error InvalidStarkNetMessaging (address starkNetMessagingAddr);
+
+    /**
+     * Invalid StarkNet UptimeFeed address.
+     * The address is 0.
+     * @param uptimeFeedAddr StarkNet UptimeFeed address.
+     */
+    error InvalidUptimeFeedAddress (uint256 uptimeFeedAddr);
+
+    /**
+     * @param starkNetMessaging the address of the StarkNet Messaging contract address
+     * @param l2UptimeFeedAddr the address of the Sequencer Uptime Feed on L2
+     */
+    constructor(address starkNetMessaging, uint256 l2UptimeFeedAddr) {
+        if (starkNetMessaging == address(0)) {
+            revert InvalidStarkNetMessaging({
+                starkNetMessagingAddr: starkNetMessaging
+            });
+        }
+
+        if (l2UptimeFeedAddr == 0) {
+            revert InvalidUptimeFeedAddress({
+                uptimeFeedAddr: l2UptimeFeedAddr
+            });
+        }
+
+        STARKNET_MESSAGING = IStarknetMessaging(starkNetMessaging);
+        L2_UPTIME_FEED_ADDR = l2UptimeFeedAddr;
+    }
+
+    /**
+     * @notice toUInt256 convert a bool to uint256.
+     */
+    function toUInt256(bool x) internal pure returns (uint256 r) {
+        assembly {
+            r := x
+        }
+    }
+
+    /**
+     * @notice versions:
+     *
+     * - StarkNetValidator 0.1.0: initial release
+     *   - now calls `updateStatus` on an L2 SequencerUptimeFeed contract instead of
+     *     directly calling the Flags contract
```

Comment:
Please remove this part:

```suggestion
     *   - now calls `updateStatus` on an L2 SequencerUptimeFeed contract instead of
     *     directly calling the Flags contract
```